src/wp-includes/widgets/class-wp-widget-archives.php

```diff
@@ -109,15 +109,31 @@ public function widget( $args, $instance ) {
 
 			<?php ob_start(); ?>
 <script>
-(function() {
-	var dropdown = document.getElementById( "<?php echo esc_js( $dropdown_id ); ?>" );
-	function onSelectChange() {
-		if ( dropdown.options[ dropdown.selectedIndex ].value !== '' ) {
-			document.location.href = this.options[ this.selectedIndex ].value;
-		}
+(( dropdownId ) => {
+	const dropdown = document.getElementById( dropdownId );
+	function onSelectChange(event) {
+		setTimeout( function() {
+			const lastKey = dropdown.getAttribute( 'data-lastkey' );
+			if ( 'change' === event.type && 'escape' === dropdown.dataset.lastkey ) {
```

Comment:
Is `'change' === event.type` needed? Won't it always be this event type?


```suggestion
			if ( 'escape' === dropdown.dataset.lastkey ) {
```

Reply:
Fixed in b0327b222083220a3fb2b78b169046d4d0c27ede